source/data-formats/working-with-bson.txt

```diff
@@ -14,386 +14,230 @@ Working with BSON
-                           << builder::stream::close_document;
-   }
-   auto after_array = in_array << builder::stream::close_array;
-
-   after_array << "another_key" << 42;
-
-   document::value doc = after_array << builder::stream::finalize;
-
-   std::cout << to_json(doc) << std::endl;
-
-.. note::
-   The result of any stream operation should be captured, so if you want
-   to split the single statement within the for loop above into multiple
-   statements, you must capture each intermediate result. Additionally, the last
-   statement within the loop body should assign its result back to the in_array
-   object, so that the loop restarts in a consistent state:
-
-   .. code-block:: cpp
-
-      for (auto && e : {1, 2, 3}) {
-         auto open_state = in_array << builder::stream::open_document;
-         auto temp_state = open_state << "key" << e;
-         in_array = temp_state << builder::stream::close_document;
-      }
-
-.. _cpp-bson-value:
-
-Owning BSON Documents
+BSON Values and Views
 ~~~~~~~~~~~~~~~~~~~~~
 
-`bsoncxx::document::value <https://github.com/mongodb/mongo-cxx-driver/blob/master/src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/document/value.hpp>`__
-
-This type represents an actual BSON document, one that owns its buffer of
-data.  These documents can be constructed from a builder by calling
-``extract()``:
-
-.. code-block:: cpp
-
-   bsoncxx::document::value basic_doc{basic_builder.extract()};
-   bsoncxx::document::value stream_doc{stream_builder.extract()};
-
-After calling ``extract()`` the builder is in a moved-from state,
-and should not be used.
-
-It's possible to create a ``bsoncxx::document::value`` in a single line using
-the stream builder interface and the ``finalize`` token. ``finalize`` returns a
-``document::value`` from a temporary stream builder:
-
-.. code-block:: cpp
-
-   // { "finalize" : "is nifty" }
-   bsoncxx::document::value one_line = bsoncxx::builder::stream::document{} << "finalize" << "is nifty" << bsoncxx::builder::stream::finalize;
-
-.. _cpp-bson-view:
-
-Non-owning BSON Documents (views)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-`bsoncxx::document::view <https://github.com/mongodb/mongo-cxx-driver/blob/master/src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/document/view.hpp>`__
-
-This type is a view into an owning ``bsoncxx::document::value``.
-
-.. code-block:: cpp
+Many {+driver-short+} methods take a BSON document as an argument.
+The ``bsoncxx`` library provides two types that represent an underlying
+BSON document: ``bsoncxx::document::value`` and ``bsoncxx:document::view``. 
 
-   bsoncxx::document::view document_view{document_value.view()};
+The ``document::value`` type represents a BSON document that
+owns its underlying buffer of data. When you pass a ``document::value`` to a
+{+driver-short+} method, the method receives a copy of the ``document::value``.
+When a ``document::value`` type goes out of scope, its underlying buffer is freed.
```

Comment:
would it be the variable that goes out of scope? or the object? i'm not very familiar with C++

Reply:
The variable, I'll make that clear